packages/polaris-viz/src/components/TooltipWrapper/utilities/getAlteredVerticalBarPosition.ts

```diff
@@ -0,0 +1,122 @@
+import {clamp} from '@shopify/polaris-viz-core';
+
+import {TooltipHorizontalOffset, TooltipVerticalOffset} from '../types';
+import type {AlteredPositionProps, AlteredPositionReturn} from '../types';
+import {
+  getAbovePosition,
+  getBelowPosition,
+  getCenterPosition,
+  getInlinePosition,
+  getLeftPosition,
+  getRightPosition,
+  getVerticalCenterPosition,
+} from '../utilities';
+import {TOOLTIP_MARGIN, SCROLLBAR_WIDTH} from '../constants';
+
+// Keep the tooltip within the bounds of the chart.
+// Based on "position" the tooltip will be placed
+// around the chart item so the item should never
+// be obscured by the tooltip.
+export function getAlteredVerticalBarPosition(
```

Comment:
We're altering the original position of the tooltip. Different charts have different rules. `BarChart` for example will move the tooltip around so that it's never covering the bar group it belongs to.